radsat-sk/main-application/comms-processing_tasks.c.c

```diff
@@ -0,0 +1,75 @@
+/*  comms-processing_tasks.c
+
+    Created by Minh Phong Bill Truong on April 28th 2020
+ */
+#include <FreeRTOS.h>
+#include <task.h>
+#include <semphr.h>
+
+#include <stdio.h>
+#include <comms-processing_tasks.h>
+
+//==============================================================================
+//                                   External
+//==============================================================================
+/**
+ * This Queue is used to store received and unprocessed packets
+ */
+QueueHandle_t rawPacketReceiveQueue;
+
+/**
+ * This Queue is used to store encoded packets that will be used by transmitTask
+ */
+QueueHandle_t packetsToTransmit;
+
+/**
+ * Used to 
+ */
+SemaphoreHandle_t encodeSignal;
+
+//==============================================================================
+//                                  FUNCTIONS
+//==============================================================================
+
+/**
+ *  @brief Call functions required to encode packet
+ *  
+ *  The encoder task receives the data that is to be sent down (clarify) and calls the
+ *  required functions to handle the formatting of the data into a packet. 
+ * 
+ *  @header	"software-and-command/main-applicatoion/comms-processing_tasks.c"
+ *	@param	Unused
+ *	@pre	None
+ * 	@post	Formatted packet is stored in memory
+ * 	@return	None
+ */
+void encodeTask( void* pvparameters ) 
+{
+    uint16_t  rawOutgoingPacket;
+    uint16_t  encodedOutgoingPacket;
+
+    while(1) 
+    {
+        if ( RAW_PACKETS_QUEUED )
+        {
+            // Wait until signal to encode packet is given.
+            xSemaphoreTake( encodeSignal, portMAX_DELAY )
+        }
+        else
+        {
+            // Execute any tasks of equal priority that are ready now.
+            taskYIELD();
```

Comment:
taskYield() won't allow other tasks of equal or lower priority to run - only higher priority. Thus, I think it would be prudent to use vTaskDelay(1U); for _most_ scenarios.